packages/BetterPhpDocParser/PhpDocManipulator/PhpDocClassRenamer.php

```diff
@@ -59,15 +60,22 @@ private function processAssertChoiceTagValueNode(array $oldToNewClasses, PhpDocI
 
         $callableCallbackArrayItems = $callbackClass->getValues();
         $classNameArrayItemNode = $callableCallbackArrayItems[0];
+        $classNameStringNode = $classNameArrayItemNode->value;
+
+        // array is needed for callable
+        if (! $classNameStringNode instanceof StringNode) {
+            return;
+        }
 
         foreach ($oldToNewClasses as $oldClass => $newClass) {
-            if ($classNameArrayItemNode->value !== $oldClass) {
+            if ($classNameStringNode->value !== $oldClass) {
                 continue;
             }
 
-            $classNameArrayItemNode->value = $newClass;
+            $classNameStringNode->value = $newClass;
 
             // trigger reprint
+//            $classNameStringNode->setAttribute(PhpDocAttributeKey::ORIG_NODE, null);
             $classNameArrayItemNode->setAttribute(PhpDocAttributeKey::ORIG_NODE, null);
```

Comment:
@TomasVotruba I don't understand quite well the "re-print" concept, for which of the two node the original node should be set to null? or both?
my guess would be that on the array item node it's enough, as it's a parent of the string node, but not sure

Reply:
This depends on the context :) when everything works perfectly, no orig node reset it needed.
If the reprint of node is not invoked, this might help.